src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -1010,16 +993,9 @@ void DataBox<tmpl::list<Tags...>>::pup_impl(
     (void)this;  // Compiler bug warns this isn't used
     using tag = decltype(current_tag);
     if (p.isUnpacking()) {
-      get_deferred<tag>() =
-          make_deferred_helper<tag>(typename tag::argument_tags{});
+      add_compute_item_to_box<tag, tmpl::list<Tags...>>();
```

Comment:
This could have a bug where a Variables is serialized and then the compute items are added that point into the Variables, followed by a deserialization of the Variables. For Variables this deserialization is likely happening as a copy into the buffer rather than a reallocation of the buffer. A class that intentionally re-allocates on deserialization should show if there is such a bug here.

Reply:
I think this is fine because the compute subitems are added as unevaluated compute items, so they will only be set up the first time they are used.  I've added a test that I think would have caught such a problem.

Reply:
Ah okay, that makes sense. Thanks!